### PR TITLE
Add Name Server reports

### PR DESCRIPTION
**Pull request by Donnype**
### Changes
Adds the basic security reports for name servers.

### Issue link

Closes part of #2081 

### Demo

image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/reports/report_types/name_server_report/report.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+@dataclass
+class NameServerCheck:
+    no_uncommon_ports: bool = False
+    has_dnssec: bool = False
+    has_valid_dnssec: bool = False
+
+
+@dataclass
+class NameServerChecks:
+    checks: List[NameServerCheck]
```
Comment (ammar92) [inline rocky/reports/report_types/name_server_report/report.py:25]: You might want to use a default factory here, such that you don't have to pass empty list literals

File: rocky/reports/report_types/name_server_report/report.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+        if reference.class_type == Hostname:
+            hostnames = [self.octopoes_api_connector.get(reference)]
+
+        elif reference.class_type in (IPAddressV4, IPAddressV6):
+            hostnames = self.octopoes_api_connector.query(
+                "Hostname.<hostname[is ResolvedHostname].address", valid_time, reference
+            )
+
+        name_server_checks = NameServerChecks(checks=[])
```
Comment (ammar92) [inline rocky/reports/report_types/name_server_report/report.py:70]: Usually not a good idea to pass mutable types as arguments; see previous point about default factories for dataclass fields

File: rocky/reports/report_types/name_server_report/report.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    @property
+    def has_dnssec(self):
+        return sum([check.has_dnssec for check in self.checks])
+
+    @property
+    def has_valid_dnssec(self):
+        return sum([check.has_valid_dnssec for check in self.checks])
+
+    def __len__(self):
+        return len(self.checks)
```
Comment (ammar92) [inline rocky/reports/report_types/name_server_report/report.py:40]: I don't see any added value for this one (except for the rest as a general wrapper with helper utils) as it seems unused, but in what context will this be used?

**Review by ammar92 — COMMENTED** [review]
Nice one. I had only one question and a suggestion for improvement

File: rocky/reports/report_types/name_server_report/report.py (hunk 3 above)
Comment (Donnype) [reply rocky/reports/report_types/name_server_report/report.py:41]: It's used in the templates

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
- Onboarding flow can be completed
- Objects and findings are created as expected
- Task and finding creation
- Report contained some minor bugs (unclear recommendation text when no DNSSEC was present, no findings were shown for IPs (0/0 name servers compliant) and some irrelevant recommendations were shown. These are all fixed now. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Comment by noamblitz** [comment]
For aggregating it would be important to know which hostnames have issues and what the ip of the system is, but we can also fix that when aggregating

**Review by dekkers — APPROVED** [review]

**Pull request merged**